src/components/docs/date-time-picker.js

```diff
@@ -0,0 +1,513 @@
+      }
+    },
+  },
+  uischema: {
+    type: 'Control',
+    scope: '#/properties/datetime',
+    options: {
+      format: 'date-time'
+    }
+  },
+  data: {
+    datetime: new Date().toISOString()
+  },
+};
+
+export const DateTimeUiSchema = () => (
+  <Demo
+    data={DateTimeUiSchemaInput.data}
+    schema={DateTimeUiSchemaInput.schema}
+    uischema={DateTimeUiSchemaInput.uischema}
+  />
+);
+
+export const DateTimeUiOptionSchemaInput = {
+  schema: {
+    properties: {
+      datetime: {
+        type: 'string',
+        description: 'uischema-based datetime picker'
+      }
+    },
+  },
+  uischema: {
+    type: 'Control',
+    scope: '#/properties/datetime',
+    options: {
+      format: 'date-time',
+      clearLabel: 'Clear it!',
+      cancelLabel: 'Abort',
+      okLabel: 'Do it',
+      dateTimeFormat: 'DD-MM-YY hh:mm:a',
+      dateTimeSaveFormat: 'YYYY/MM/DD h:mm a',
+      ampm: true
+    }
+  },
+  data: {
+    datetime: new Date().toISOString()
+  },
+};
+
+export const DateTimeUiOptionSchema = () => (
+  <Demo
+    data={DateTimeUiOptionSchemaInput.data}
+    schema={DateTimeUiOptionSchemaInput.schema}
+    uischema={DateTimeUiOptionSchemaInput.uischema}
+  />
+);
+
+
+export const DateSchemaGerman = {
+  schema: {
+    properties: {
+    },
+  },
+  uischema: {
+    type: 'Control',
+    scope: '#/properties/date',
+    label: 'Year Month Picker',
+    options: {
+      format: 'date',
+      clearLabel: 'Zurücksetzen',
+      cancelLabel: 'Abbrechen',
+      okLabel: 'Ok'
+    },
+  },
+  data: {
+    date: new Date().toISOString().substr(0, 10),
+  },
+};
+
+function sleep (time) {
+  return new Promise((resolve) => setTimeout(resolve, time));
+}
+
+
+const ReloadJsonForms = ({reload,data}) => {
+  sleep(50).then(() => {
+    reload();
+  });
+  return <JsonForms
+            data={data}
+            schema={DateSchemaGerman.schema}
+            uischema={DateSchemaGerman.uischema}
+            renderers={materialRenderers}
+            cells={materialCells}
+            onChange={({ data, _errors }) => {}}
+          />
+}
+
+export const LocalizationExample = () => {
+  
+  const [data, setData] = useState(DateSchemaInput.data);
+  const [local, setLocal] = useState('en');
+  const [reloadJsForms, setreloadJsForms] = useState(false);
+
+  dayjs.locale(local);
+
+  return (
+    <div style={{padding:"8px"}}>
+    <Grid
+      container
+      direction='column'
+      justifyContent='space-between'
+      alignItems="stretch"
+      >
+        <Select 
+            value = {local}
+            onChange={(event) =>{
+              setreloadJsForms(true);
+              setLocal(event.target.value);
+            }}
+            style={{marginBottom:'20px'}}
+            >
+            <MenuItem value={'en'}>en</MenuItem>
+            <MenuItem value={'de'}>de</MenuItem>
+          </Select>
+          {reloadJsForms===true?<ReloadJsonForms data = {data}reload={()=>{setreloadJsForms(false)}}/>:
```

Comment:
To change the language, it is not enough to rerender this component or to render JsonForms with different probs, because some components don’t get rerendered/updated. I removed Jsonforms from the DOM just to add it again, this way everything gets updated.